### ISSUE TITLE
Cannot use #create_block_blob method with IO/File object

### ISSUE THREAD

**Issue by rtyler** [Azure/azure-storage-ruby#61]
The call to `#encoding` only exists on `String` like objects`

```
/home/tyler/.rvm/gems/ruby-2.2.0/gems/azure-storage-0.11.3.preview/lib/azure/storage/blob/blob_service.rb:50:in `call': undefined method `encoding' for #<File:/srv/releases/jenkins/debian/jenkins_2.29_all.deb> (NoMethodError)
        from /home/tyler/.rvm/gems/ruby-2.2.0/gems/azure-storage-0.11.3.preview/lib/azure/storage/blob/block.rb:114:in `create_block_blob'
        from blob-sync:44:in `block (2 levels) in <main>'
        from /home/tyler/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/set.rb:283:in `each_key'
        from /home/tyler/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/set.rb:283:in `each'
        from blob-sync:39:in `block in <main>'
        from blob-sync:11:in `each'
        from blob-sync:11:in `<main>'
```

**Comment by yaxia** [comment]
@rtyler  Thanks for reporting this issue. We will fix it in a coming release.

**Comment by yaxia** [comment]
It is fixed in 0.11.4. Please take a try. Thanks.
